### DISCUSSION TITLE
NEXT.js ISR remove static page

### DISCUSSION THREAD

**Discussion by meerbahadin** [vercel/community#1366] [Help] [answered]
im experiencing this issue anyone knows how to fix it ?

the original question from stackoverflow:
https://stackoverflow.com/questions/66930552/next-js-isr-remove-static-page

basically I want remove statically generated page after deleting it from CMS because now the page still exist

**Comment by mavr1k** [comment]
Does somebody take a look at that? 
We have data stored in the DB and want to remove cached pages from the ISR mechanisms when it have been removed from the DB.

**Comment by amyegan** [comment, marked as answer]
You can use [On-demand Revalidation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation) to update or remove old content as needed. 

An example was shared in the Stack Overflow post: [stackoverflow.com/a/73730647](https://stackoverflow.com/a/73730647)

**Reply by mavr1k** [reply]
I'm using that, by calling `revalidate` in the Edge API. However, the document doesn't remove from the cache.

**Reply by laurenskling** [reply]
I have the same question. A removed page returns a 404, but as stated in: https://nextjs.org/docs/app/building-your-application/caching#time-based-revalidation `If the background revalidation fails, the previous data will be kept unaltered.`. A 404 is a fail. So `fetch` will returned the unaltered cached page, not a 404 page..
